### PR TITLE
DOC: Added sentence to docstring of histogram_bin_edges to explain bin width

### PR DESCRIPTION
**Pull request by jamiebarker0310**
This regards #18319 where it was discussed that the width of the bins returned is not equal to what would is described by the formulas described in the "bins" parameter.

### CODE REVIEW

**Comment by melissawm** [comment]
I think the build failure is unrelated, builds fine locally.

**Comment by seberg** [comment]
Yeah, we are seeing that a lot just now:
```
make: *** [Makefile:179: html-build] Terminated

Too long with no output (exceeded 10m0s): context deadline exceeded
```
I hopened gh-18349, from the docs it sounds like that should fix it (https://support.circleci.com/hc/en-us/articles/360007188574-Build-has-hit-timeout-limit)

File: numpy/lib/histograms.py [hunk 1]
```diff
@@ -562,7 +562,9 @@ def histogram_bin_edges(a, bins=10, range=None, weights=None):
     below, :math:`h` is the binwidth and :math:`n_h` is the number of
     bins. All estimators that compute bin counts are recast to bin width
     using the `ptp` of the data. The final bin count is obtained from
-    ``np.round(np.ceil(range / h))``.
+    ``np.round(np.ceil(range / h))``. The final bin width is often less 
+    than what is returned by the formulas below, in order to allow the
```
Comment (madphysicist) [inline numpy/lib/histograms.py:566]: Suggestion: "The final bin width will therefore generally be less than what is returned by the estimators." "estimator" is a better term than formula, and more consistent. The part about entire range is covered by `ptp` two sentences back.

Comment (jamiebarker0310) [reply numpy/lib/histograms.py:566]: Thanks for your comment, will adjust and repush

**Review by madphysicist — APPROVED** [review]
I think this is a good change. Hopefully it gets included soon.

**Comment by melissawm** [comment]
Thanks, @jamiebarker0310 !

**Pull request merged**